v03_pipeline/lib/misc/family_loading_failures.py

```diff
@@ -10,83 +12,92 @@ def passes_relatedness_check(
     sample_id: str,
     other_id: str,
     relation: Relation,
-) -> bool:
+) -> tuple[bool, str | None]:
     # No relationship to check, return true
     if other_id is None:
-        return True
+        return True, None
     coefficients = relatedness_check_lookup.get(
         (min(sample_id, other_id), max(sample_id, other_id)),
     )
-    if not coefficients:
-        return False
-    return np.allclose(
+    if not coefficients or not np.allclose(
         coefficients,
         relation.coefficients,
         0.1,
-    )
+    ):
+        return (
+            False,
+            f'Sample {sample_id} has expected relation "{relation.value}" to {other_id} but has coefficients {coefficients or []}',
+        )
+    return True, None
 
 
 def passes_all_relatedness_checks(  # noqa: C901
     relatedness_check_lookup: dict[tuple[str, str], list],
     sample: Sample,
-) -> bool:
+) -> tuple[bool, str | None]:
```

Comment:
aw, I just noticed that this is still written such that only the first failure will be returned, even though we have the ability to return all relationship failures.  I will quickly fix that!